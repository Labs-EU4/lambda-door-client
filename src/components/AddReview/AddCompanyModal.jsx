/* eslint-disable react/no-array-index-key */
import React from 'react';
import { connect } from 'react-redux';
import { withFormik } from 'formik';
import * as yup from 'yup';
import { Modal, Form, Input, Icon, Button, Select } from 'antd';
import { addCompany } from '../../state/actions/companies';
import LocationSearch from '../LocationSearch';
import openNotification from '../../utils/openNotification';

const { Option } = Select;

const AddCompanyModal = props => {
  const {
    visible,
    setAddingCompany,
    values,
    handleChange,
    handleBlur,
    handleSubmit,
    touched,
    errors,
  } = props;

  const handleClose = () => {
    setAddingCompany(false);
  };

  return (
    <Modal
      title="Add A Company"
      visible={visible}
      onCancel={handleClose}
      footer={[
        <Button key="back" onClick={handleClose}>
          Cancel
        </Button>,
      ]}
    >
      <form onSubmit={handleSubmit}>
        <Form.Item
          help={touched.name && errors.name ? errors.name : ''}
          validateStatus={touched.name && errors.name ? 'error' : undefined}
        >
          <Input
            size="large"
            name="name"
            value={values.name}
            onChange={handleChange}
            onBlur={handleBlur}
            placeholder="Company Name"
            prefix={<Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />}
          />
        </Form.Item>

        <Form.Item
          help={touched.location && errors.location ? errors.location : ''}
          validateStatus={
            touched.location && errors.location ? 'error' : undefined
          }
        >
          <LocationSearch
            name="location"
            value={values.location}
            onChange={handleChange}
            onBlur={handleBlur}
            placeholder="Location"
            prefix={<Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />}
          />
        </Form.Item>
        <Form.Item
          help={touched.website && errors.website ? errors.website : ''}
          validateStatus={
            touched.website && errors.website ? 'error' : undefined
          }
        >
          <Input
            size="large"
            name="website"
            value={values.website}
            onChange={handleChange}
            onBlur={handleBlur}
            placeholder="Website"
            prefix={<Icon type="global" style={{ color: 'rgba(0,0,0,.25)' }} />}
          />
        </Form.Item>
        <Form.Item
          help={touched.type && errors.type ? errors.type : ''}
          validateStatus={touched.type && errors.type ? 'error' : undefined}
        >
          <Select
            name="type"
            value={values.type}
            placeholder="Company Type"
            onChange={value => {
              const event = { ...document.createEvent('Event') };
              event.target = {};
              event.target.name = 'type';
              event.target.value = value;
              handleChange(event);
            }}
            onBlur={handleBlur}
          >
            {[
              'Company - Public',
              'Company - Private',
              'Contract',
              'Franchise',
              'Subsidiary or Business Segment',
              'Hospital',
              'Private Practice',
              'School',
              'College',
              'Government',
              'Self Employed',
              'Other',
            ].map((elem, idx) => {
              return (
                <Option key={idx} value={elem}>
                  {elem}
                </Option>
              );
            })}
          </Select>
        </Form.Item>

        <Form.Item
          help={
            touched.description && errors.description ? errors.description : ''
          }
          validateStatus={
            touched.description && errors.description ? 'error' : undefined
          }
        >
          <Input.TextArea
            size="large"
            name="description"
            value={values.description}
            onChange={handleChange}
            onBlur={handleBlur}
            placeholder="Description"
            prefix={<Icon type="book" style={{ color: 'rgba(0,0,0,.25)' }} />}
          />
        </Form.Item>
        <Button htmlType="submit" type="primary">
          Add Company
        </Button>
      </form>
    </Modal>
  );
};

const validationSchema = yup.object().shape({
  name: yup.string().required('Please provide a name'),
  location: yup.string().required('Please provide a location'),
  website: yup
    .string()
    .url('Please enter a valid URL')
    .required('Please provide a website'),
  type: yup.string().required('Please provide a company type'),
  description: yup.string().required('Please provide a description'),
});

const CompanyModal = withFormik({
  mapPropsToValues: props => {
    return {
      name: props.value,
      location: '',
      website: '',
      description: '',
      latitude: '',
      longitude: '',
    };
  },
  handleSubmit: (values, { props, setSubmitting }) => {
    props.addCompany(values);
    props.setAddingCompany(false);
    openNotification('Company Added Successfully!');
    setSubmitting(false);
  },
  validationSchema: validationSchema,
  enableReinitialize: true,
})(AddCompanyModal);

export default connect(state => state, { addCompany })(CompanyModal);
