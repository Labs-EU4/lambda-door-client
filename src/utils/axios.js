import axios from 'axios';

axios.defaults.baseURL = `${process.env.REACT_APP_BACKEND_URL}`

export default function withAuth() {
  const token = localStorage.getItem('token');
  const instance = axios.create({
    headers: {
      'Content-Type': 'application/json',
      Authorization: `${token}`,
    },
  });

  return instance;
}
