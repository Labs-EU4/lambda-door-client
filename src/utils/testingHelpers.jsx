import React from 'react';
import { createStore, applyMiddleware, compose } from 'redux';
import { Provider } from 'react-redux';
import { Router, Route } from 'react-router-dom';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import rootReducer from '../state/reducers/index';

export const TestingRouter = ({
  ComponentWithRedirection,
  RedirectUrl,
  MainRoute,
}) => (
  <>
    <Route
      path={MainRoute}
      exact={true}
      render={props => {
        return <ComponentWithRedirection {...props} />;
      }}
    />
    <Route path={RedirectUrl} render={() => <div>{RedirectUrl}</div>} />
  </>
);

export const renderWithRedux = (
  component,
  {
    initialState,
    store = createStore(
      rootReducer,
      initialState,
      compose(applyMiddleware(thunk))
    ),
  } = {},
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>{component}</Router>
      </Provider>
    ),
    store,
    history,
  };
};
