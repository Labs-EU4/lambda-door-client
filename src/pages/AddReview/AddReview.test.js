import React from 'react';
import * as rtl from '@testing-library/react';
import AddReview from './AddReview';
import '@testing-library/jest-dom/extend-expect';
import { renderWithRedux } from '../../utils/testingHelpers';

beforeEach(rtl.cleanup);

describe('AddReview', () => {
  it('renders without crashing', () => {
    renderWithRedux(<AddReview />);
  });
  it('renders correctly', () => {
    expect(renderWithRedux(<AddReview />).baseElement).toMatchSnapshot();
  });
  it('renders correctly', () => {
    // expect().queryByText(
    //   /Pick currency/i
    // );
    // expect().quer;
    // const { render } = renderWithRedux(<AddReview />).baseElement;
    // console.log(render);
    // console.log(renderWithRedux(<AddReview />).baseElement);
  });
});
