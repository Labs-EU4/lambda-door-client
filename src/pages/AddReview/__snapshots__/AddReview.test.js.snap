// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReview renders correctly 1`] = `
<body>
  <div>
    <div>
      <h1>
        <b>
          Add your Review
        </b>
      </h1>
      <br />
      <form
        class="ant-form ant-form-vertical"
      >
        <h3>
          <b>
            General Review
          </b>
        </h3>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control"
            >
              <span
                class="ant-form-item-children"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      title="Company Name"
                    >
                      Company Name
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control-wrapper"
                  >
                    <div
                      class="ant-form-item-control"
                    >
                      <span
                        class="ant-form-item-children"
                      >
                        <div
                          class="certain-category-search ant-select-show-search ant-select-auto-complete ant-select ant-select-combobox ant-select-enabled"
                          style="width: 100%;"
                        >
                          <div
                            aria-autocomplete="list"
                            aria-controls="test-uuid"
                            aria-expanded="false"
                            aria-haspopup="true"
                            class="ant-select-selection
            ant-select-selection--single"
                            role="combobox"
                          >
                            <div
                              class="ant-select-selection__rendered"
                            >
                              <ul>
                                <li
                                  class="ant-select-search ant-select-search--inline"
                                >
                                  <div
                                    class="ant-select-search__field__wrap"
                                  >
                                    <input
                                      class="ant-input ant-select-search__field"
                                      placeholder="Company name"
                                      type="text"
                                      value=""
                                    />
                                    <span
                                      class="ant-select-search__field__mirror"
                                    >
                                      
                                      Â 
                                    </span>
                                  </div>
                                </li>
                              </ul>
                            </div>
                            <span
                              class="ant-select-arrow"
                              style="user-select: none;"
                              unselectable="on"
                            >
                              <i
                                aria-label="icon: down"
                                class="anticon anticon-down ant-select-arrow-icon"
                              >
                                <svg
                                  aria-hidden="true"
                                  class=""
                                  data-icon="down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                  />
                                </svg>
                              </i>
                            </span>
                          </div>
                        </div>
                      </span>
                    </div>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class=""
              title="Overall Rating"
            >
              Overall Rating
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control"
            >
              <span
                class="ant-form-item-children"
              >
                <ul
                  class="ant-rate"
                  role="radiogroup"
                  tabindex="0"
                >
                  <li
                    class="ant-rate-star ant-rate-star-zero"
                  >
                    <div
                      aria-checked="false"
                      aria-posinset="1"
                      aria-setsize="5"
                      role="radio"
                      tabindex="0"
                    >
                      <div
                        class="ant-rate-star-first"
                      >
                        <i
                          aria-label="icon: star"
                          class="anticon anticon-star"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="star"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                            />
                          </svg>
                        </i>
                      </div>
                      <div
                        class="ant-rate-star-second"
                      >
                        <i
                          aria-label="icon: star"
                          class="anticon anticon-star"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="star"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                            />
                          </svg>
                        </i>
                      </div>
                    </div>
                  </li>
                  <li
                    class="ant-rate-star ant-rate-star-zero"
                  >
                    <div
                      aria-checked="false"
                      aria-posinset="2"
                      aria-setsize="5"
                      role="radio"
                      tabindex="0"
                    >
                      <div
                        class="ant-rate-star-first"
                      >
                        <i
                          aria-label="icon: star"
                          class="anticon anticon-star"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="star"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                            />
                          </svg>
                        </i>
                      </div>
                      <div
                        class="ant-rate-star-second"
                      >
                        <i
                          aria-label="icon: star"
                          class="anticon anticon-star"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="star"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                            />
                          </svg>
                        </i>
                      </div>
                    </div>
                  </li>
                  <li
                    class="ant-rate-star ant-rate-star-zero"
                  >
                    <div
                      aria-checked="false"
                      aria-posinset="3"
                      aria-setsize="5"
                      role="radio"
                      tabindex="0"
                    >
                      <div
                        class="ant-rate-star-first"
                      >
                        <i
                          aria-label="icon: star"
                          class="anticon anticon-star"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="star"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                            />
                          </svg>
                        </i>
                      </div>
                      <div
                        class="ant-rate-star-second"
                      >
                        <i
                          aria-label="icon: star"
                          class="anticon anticon-star"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="star"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                            />
                          </svg>
                        </i>
                      </div>
                    </div>
                  </li>
                  <li
                    class="ant-rate-star ant-rate-star-zero"
                  >
                    <div
                      aria-checked="false"
                      aria-posinset="4"
                      aria-setsize="5"
                      role="radio"
                      tabindex="0"
                    >
                      <div
                        class="ant-rate-star-first"
                      >
                        <i
                          aria-label="icon: star"
                          class="anticon anticon-star"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="star"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                            />
                          </svg>
                        </i>
                      </div>
                      <div
                        class="ant-rate-star-second"
                      >
                        <i
                          aria-label="icon: star"
                          class="anticon anticon-star"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="star"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                            />
                          </svg>
                        </i>
                      </div>
                    </div>
                  </li>
                  <li
                    class="ant-rate-star ant-rate-star-zero"
                  >
                    <div
                      aria-checked="false"
                      aria-posinset="5"
                      aria-setsize="5"
                      role="radio"
                      tabindex="0"
                    >
                      <div
                        class="ant-rate-star-first"
                      >
                        <i
                          aria-label="icon: star"
                          class="anticon anticon-star"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="star"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                            />
                          </svg>
                        </i>
                      </div>
                      <div
                        class="ant-rate-star-second"
                      >
                        <i
                          aria-label="icon: star"
                          class="anticon anticon-star"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="star"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                            />
                          </svg>
                        </i>
                      </div>
                    </div>
                  </li>
                </ul>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control"
            >
              <span
                class="ant-form-item-children"
              >
                <div
                  class="sc-bdVaJa iCMYSg"
                >
                  <div>
                    <p>
                      I am a current employee
                    </p>
                    <button
                      aria-checked="false"
                      class="ant-switch"
                      name="is_currently_employed"
                      role="switch"
                      type="button"
                    >
                      <span
                        class="ant-switch-inner"
                      >
                        <i
                          aria-label="icon: close"
                          class="anticon anticon-close"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="close"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                            />
                          </svg>
                        </i>
                      </span>
                    </button>
                  </div>
                  <div>
                    <p>
                      I am accepting more questions
                    </p>
                    <button
                      aria-checked="false"
                      class="ant-switch"
                      name="is_accepting_questions"
                      role="switch"
                      type="button"
                    >
                      <span
                        class="ant-switch-inner"
                      >
                        <i
                          aria-label="icon: close"
                          class="anticon anticon-close"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="close"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                            />
                          </svg>
                        </i>
                      </span>
                    </button>
                  </div>
                  <div>
                    <p>
                      I want to be anonymous
                    </p>
                    <button
                      aria-checked="false"
                      class="ant-switch"
                      role="switch"
                      type="button"
                    >
                      <span
                        class="ant-switch-inner"
                      >
                        <i
                          aria-label="icon: close"
                          class="anticon anticon-close"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="close"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                            />
                          </svg>
                        </i>
                      </span>
                    </button>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class=""
              title="Review"
            >
              Review
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control"
            >
              <span
                class="ant-form-item-children"
              >
                <textarea
                  class="ant-input"
                  maxlength="255"
                  minlength="5"
                  name="review"
                  placeholder="Please share some of the pros and cons of working at this company"
                  required=""
                  rows="10"
                />
              </span>
            </div>
          </div>
        </div>
        <h3>
          <b>
            Salary Review
          </b>
        </h3>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class=""
              title="Job Title"
            >
              Job Title
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control"
            >
              <span
                class="ant-form-item-children"
              >
                <input
                  class="ant-input"
                  name="job_title"
                  placeholder="Job Title"
                  type="text"
                  value=""
                />
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class=""
              title="Employment Type"
            >
              Employment Type
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control"
            >
              <span
                class="ant-form-item-children"
              >
                <div
                  class="ant-select ant-select-enabled"
                >
                  <div
                    aria-autocomplete="list"
                    aria-controls="test-uuid"
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="ant-select-selection
            ant-select-selection--single"
                    role="combobox"
                    tabindex="0"
                  >
                    <div
                      class="ant-select-selection__rendered"
                    >
                      <div
                        class="ant-select-selection__placeholder"
                        style="display: block; user-select: none;"
                        unselectable="on"
                      >
                        Employment Type
                      </div>
                    </div>
                    <span
                      class="ant-select-arrow"
                      style="user-select: none;"
                      unselectable="on"
                    >
                      <i
                        aria-label="icon: down"
                        class="anticon anticon-down ant-select-arrow-icon"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </i>
                    </span>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class=""
              title="Job Description"
            >
              Job Description
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control"
            >
              <span
                class="ant-form-item-children"
              >
                <textarea
                  class="ant-input"
                  maxlength="255"
                  minlength="5"
                  name="description"
                  placeholder="Please share with us what the job role involves"
                  required=""
                  rows="5"
                />
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class=""
              title="Salary"
            >
              Salary
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control"
            >
              <span
                class="ant-form-item-children"
              >
                <div
                  style="display: flex; width: 80%; justify-content: space-between;"
                >
                  <div>
                    <input
                      class="ant-input"
                      name="unit"
                      placeholder="Annual Salary"
                      step="0.01"
                      type="number"
                      value=""
                    />
                  </div>
                  <div
                    class="currency"
                    style="width: 60%;"
                  >
                    <div
                      class="ant-select ant-select-enabled"
                    >
                      <div
                        aria-autocomplete="list"
                        aria-controls="test-uuid"
                        aria-expanded="false"
                        aria-haspopup="true"
                        class="ant-select-selection
            ant-select-selection--single"
                        role="combobox"
                        tabindex="0"
                      >
                        <div
                          class="ant-select-selection__rendered"
                        >
                          <div
                            class="ant-select-selection__placeholder"
                            style="display: block; user-select: none;"
                            unselectable="on"
                          >
                            Pick currency
                          </div>
                        </div>
                        <span
                          class="ant-select-arrow"
                          style="user-select: none;"
                          unselectable="on"
                        >
                          <i
                            aria-label="icon: down"
                            class="anticon anticon-down ant-select-arrow-icon"
                          >
                            <svg
                              aria-hidden="true"
                              class=""
                              data-icon="down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              />
                            </svg>
                          </i>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
        <h3>
          <b>
            Interview Process Review
          </b>
        </h3>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class=""
              title="Review"
            >
              Review
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control"
            >
              <span
                class="ant-form-item-children"
              >
                <textarea
                  class="ant-input"
                  maxlength="255"
                  minlength="5"
                  name="text"
                  placeholder="Please share the steps involved in the interview process"
                  required=""
                  rows="10"
                />
              </span>
            </div>
          </div>
        </div>
        <button
          class="ant-btn ant-btn-primary"
          disabled=""
          type="submit"
        >
          <span>
            Submit
          </span>
        </button>
      </form>
    </div>
  </div>
</body>
`;
